hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -769,8 +773,12 @@ public DirectoryInfo(String relativePath, FileStatus[] fileStatus) {
         } else if (status.getPath().getName().equals(HoodiePartitionMetadata.HOODIE_PARTITION_METAFILE)) {
           // Presence of partition meta file implies this is a HUDI partition
           this.isHoodiePartition = true;
-        } else if (FSUtils.isDataFile(status.getPath())) {
-          // Regular HUDI data file (base file or log file)
+        } else if (FSUtils.isBaseFile(status.getPath())) {
+          final String commitTime = FSUtils.getCommitTime(status.getPath().toString());
+          if (completedInstantsTimeline.containsOrBeforeTimelineStarts(commitTime)) {
+            filenameToSizeMap.put(status.getPath().getName(), status.getLen());
+          }
```

Comment:
can you leave a comment here as to why we do this

Reply:
I am taking on this new functionality change in a new PR